server/manager/src/main/java/org/apache/accumulo/manager/recovery/RecoveryManager.java

```diff
@@ -156,72 +156,68 @@ private boolean exists(final Path path) throws IOException {
     }
   }
 
-  public boolean recoverLogs(KeyExtent extent, Collection<Collection<String>> walogs)
-      throws IOException {
+  public boolean recoverLogs(KeyExtent extent, Collection<LogEntry> walogs) throws IOException {
     boolean recoveryNeeded = false;
 
-    for (Collection<String> logs : walogs) {
-      for (String walog : logs) {
-
-        Path switchedWalog = VolumeUtil.switchVolume(new Path(walog), FileType.WAL,
```

Comment:
Do you know if this removed the need for  FileType.WAL?  Nice that this could use types instead of the enum.

Reply:
It didn't. There just happened to already be an overloaded version of this method that was for logs, so I called that to reduce the boilerplate. FileType.WAL is still used inside the overloaded method. The only other thing inside the overloaded method is some extra trace logging, which I didn't mind picking up if I could reduce some boilerplate.